openbb_platform/platform/provider/openbb_provider/standard_models/calendar_ipo.py

```diff
@@ -0,0 +1,66 @@
+"""IPO Calendar  data model."""
+
+from datetime import (
+    date as dateType,
+    datetime,
+)
+from typing import Optional
+
+from pydantic import Field, field_validator
+
+from openbb_provider.abstract.data import Data
+from openbb_provider.abstract.query_params import QueryParams
+
+
+class CalendarIpoQueryParams(QueryParams):
+    """IPO Calendar Query Params."""
+
+    symbol: Optional[str] = Field(
+        description="Return IPOs with the given ticker (typically the IPO for the company).",
+        default=None,
+    )
+    start_date: Optional[dateType] = Field(
+        description="Return IPOs on or after the given date.",
+        default=None,
+    )
+    end_date: Optional[dateType] = Field(
+        description="Return IPOs on or before the given date.", default=None
+    )
+    limit: Optional[int] = Field(
+        description="Limit the number of results returned by most recent.",
+        default=300,
+    )
+
+    @field_validator("symbol", mode="before", check_fields=False)
+    def upper_symbol(cls, v: str):
+        """Convert symbol to uppercase."""
+        return v.upper() if v else None
+
+    @field_validator("start_date", mode="before", check_fields=False)
+    def start_date_validate(cls, v):  # pylint: disable=E0213
+        """Return the date as a datetime object."""
+        return datetime.strftime(v, "%Y-%m-%d") if v else None
+
+    @field_validator("end_date", mode="before", check_fields=False)
+    def end_date_validate(cls, v):  # pylint: disable=E0213
+        """Return the date as a datetime object."""
+        return datetime.strftime(v, "%Y-%m-%d") if v else None
+
+
+class CalendarIpoData(Data):
+    """IPO Calendar Data."""
+
+    date: Optional[dateType] = Field(
```

Comment:
And then we can call this ipo_date to be more desciptive.

Reply:
So the reason to use `date` is that when `.to_df()` is called it is indexed to that column.  AFAIK, that's the only way to do this now because `to_df()` does not accept any params.

Reply:
> Or allow to_df/to_dataframe to take an index.

This is the better alternative, then you could choose to not index the column when there is one labelled as `date`.